pkg/promlib/resource/resource.go

```diff
@@ -173,8 +173,8 @@ func (r *Resource) GetSuggestions(ctx context.Context, req *backend.CallResource
 		values.Add("match[]", vs.String())
 	}
 
-	// if no timeserie name is provided, but scopes are, the scope is still rendered and passed as match param.
-	if len(selectorList) == 0 && len(sugReq.Scopes) > 0 {
+	// if no timeserie name is provided, but scopes or adhoc filters are, the scope is still rendered and passed as match param.
+	if len(selectorList) == 0 && (len(sugReq.Scopes) > 0 || len(sugReq.AdhocFilters) > 0) {
```

Comment:
we should probably test the `matchers` directly, as that's input we're interested in.

so something like this:

```
        if len(selectorList) == 0 && len(matchers > 0) {
```

related, but not required, we seem to do work we don't need to - if there are any selectorList values, we don't need to create matchers.